src/java.base/share/classes/com/sun/crypto/provider/GHASH.java

```diff
@@ -198,6 +199,38 @@ void update(byte[] in, int inOfs, int inLen) {
         processBlocks(in, inOfs, inLen/AES_BLOCK_SIZE, state, subkeyHtbl);
     }
 
+    // Maximum buffer size rotating ByteBuffer->byte[] intrinsic copy
+    private static final int MAX_LEN = 1024;
+
+    // Will process as many blocks it can and will leave the remaining.
+    int update(ByteBuffer src, int inLen) {
+        if (inLen == 0) {
+            return 0;
+        }
+
+        inLen -= (inLen % AES_BLOCK_SIZE);
```

Comment:
Maybe the if-(inLen == 0) check should be moved down after this line to cover cases where inLen < AES_BLOCK_SIZE?

Reply:
Sure that makes sense.